Block/NodeType/Category.php

```diff
@@ -150,6 +159,24 @@ public function getCategoryUrl($nodeId, $storeId = null)
         return false;
     }
 
+    /**
+     * @param int $nodeId
+     *
+     * @return CategoryRepository
+     */
+    public function getCategory(int $nodeId)
+    {
+        if (!isset($this->nodes[$nodeId])) {
+            throw new \InvalidArgumentException('Invalid node identifier specified');
+        }
+
+        $storeId = $this->_storeManager->getStore()->getId();
+        $node = $this->nodes[$nodeId];
+        $categoryId = (int) $node->getContent();
+
+        return $this->categoryRepository->get($categoryId, $storeId);
```

Comment:
Do you load a category in a loop here (which renders a single node)? I think isn't the best idea. Can we load all needed categories as a collection before rendering our menu?

Reply:
Done